### PR TITLE
Update CMakeUserPresets.TEMPLATE.json

### PR DESCRIPTION
**Pull request by networkfusion**
## Description
Improves the comments to make it easier for new users to understand.

## Motivation and Context
The options did not make it clear that they are optional.
It also wraps them (as non conforment) json comments, as it will (possibily) work on some systems and most users will recognise them as such.

## Types of changes
- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist
- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

**Review by josesimoes — COMMENTED** [review]
All of them are optional...
And they can be removed or set to `null` (useful if one wants to use them intermittently).

As for the comments prefix & suffix... Whatever pleases you... 😁 They are inside double quotes so anything goes.

**Comment by networkfusion** [comment]
> All of them are optional... And they can be removed or set to `null` (useful if one wants to use them intermittently).
> 
> As for the comments prefix & suffix... Whatever pleases you... 😁 They are inside double quotes so anything goes.

Agreed, but I am just trying to make it clear (to new users) that they are indeed optional. Perhaps I should move the `quotes` inside the `comment`...

**Comment by josesimoes** [comment]
Understod. So please have them all start with optional.

But the last part, I'm concerned that it maybe misleading because they can perfectly be there, just set to null. That's described in the documentation.

If you want to have these manageable by a script they are better shorter than longer.

Also, whatever changes you make there do not make it break the json spec. This file it's used in the pipeline and if it's not compliant we'll have to fix it or remedy afterwards.

**Comment by networkfusion** [comment]
> Understod. So please have them all start with optional.
> 
> But the last part, I'm concerned that it maybe misleading because they can perfectly be there, just set to null. That's described in the documentation.
> 
> If you want to have these manageable by a script they are better shorter than longer.
> 
> Also, whatever changes you make there do not make it break the json spec. This file it's used in the pipeline and if it's not compliant we'll have to fix it or remedy afterwards.

So, that means keeping them within the quotes (as current).
Perhaps changing them to:
`"<!--OPTIONAL-SEE-README-FOR-DETAILS-->"`

**Comment by networkfusion** [comment]
And (or) possibily adding the desctiption as verbose, so...
" reason here"

**Comment by josesimoes** [comment]
That's concise for sure. The difference is that before each one was kind of self explanatory of what was meant to be there. Now it makes a new comer to really go to the doc to know more about eahx one.

**Comment by networkfusion** [comment]
Hopefully I have found a balance now...

File: CMakeUserPresets.TEMPLATE.json [hunk 1]
```diff
@@ -3,24 +3,25 @@
...
-                "STM32_CUBE_PACKAGE_SOURCE": "<path-to-stm32-cubemx-package-mind-the-forward-slash>",
-                "MBEDTLS_SOURCE": "<path-to-mbedtls-source-mind-the-forward-slashes>",
-                "FATFS_SOURCE": "<path-to-local-FATFS-source-mind-the-forward-slash>",
-                "SPIFFS_SOURCE": "<path-to-local-SPIFFS-source-mind-the-forward-slash>",
-                "ESP32_IDF_PATH": "<absolute-path-to-the-IDF-folder-mind-the-forward-slashes>",
-                "TI_SL_CC32xx_SDK_SOURCE": "<path-to-local-TI_SimpleLink-CC32xx-SDK-source-mind-the-forward-slash>",
-                "TI_SL_CC13xx_26xx_SDK_SOURCE": "<path-to-local-TI_SimpleLink-CC13xx_26xx-SDK-source-mind-the-forward-slash>",
-                "TI_XDCTOOLS_SOURCE": "<path-to-local-TI-XDC-Tools-source-mind-the-forward-slash>",
-                "TI_SYSCONFIG_SOURCE": "<path-to-local-TI-SYSCONFIG-Tool-source-mind-the-forward-slash>"
+                "USER_PRESET_DEFAULT_WARNING_NOTICE": "Update variables and ensure the use of forward slashes in all paths",
```
Comment (networkfusion) [inline CMakeUserPresets.TEMPLATE.json:9]: Note, does Cmake Presets have this hangover of `/`

**Comment by networkfusion** [comment]
A further improvement would be to `MATCH` https://cmake.org/cmake/help/latest/command/string.html the start of the string when calling them from the default and warning the user that they are default. If they are default, we could "choose" to continue...

**Review by josesimoes — COMMENTED** [review]
I'm not seeing this making it easier to new comers... 
When I look at this doc (the full doc), all the repetition with (OPTIONAL: SEE-README-FOR-DETAILS) increases noise and makes it harder to get to the relevant information that's to be conveyed: what is intended to be on each option, if it's going to be used.

Again: if we feel relevant to pass the message that each option is optional (I'm fine with that) we can do that more elegantly in the description field.
If we need to point to the readme, we can add that in the description field, even adding the URL for the real doc.

**Comment by networkfusion** [comment]
Ok. I think I will close this for the moment. If a better idea comes along we will act accordingly.

**Comment by josesimoes** [comment]
Don't throw it all away! It's OK to improve all that in the description. Like mentioning that they are all optional, the extra details on the null thing and adding the URL for the doc explaining the various options. _me thinks..._  😁

**Pull request closed without merging**